src/gateway/get-started/key-authentication.md

```diff
@@ -0,0 +1,223 @@
+   by sending a `POST` request to the plugins object on the Admin API:
+
+   ```sh
+   curl -X POST http://localhost:8001/plugins/ \
+       --data "name=key-auth"  \
+       --data "config.key_names=apikey"
+   ```
+
+   You will receive a `201` response indicating the plugin was installed.
+
+   The `key_names` configuration field in the above request defines the name of the field that the
+   plugin will look for to read the key when authenticating requests. The plugin will look for the field in headers,
+   query string parameters, and the request body.
+  
+1. **Send an unauthenticated request** 
+
+   Try to access the service without providing the key:
+   
+   ```sh
+   curl -i http://localhost:8000/mock/request
+   ```
+   
+   Since you enabled key authentication globally, you will receive an unauthorized response:
+   
+   ```text
+   HTTP/1.1 401 Unauthorized
+   ...
+   {
+       "message": "No API key found in request"
+   }
+   ```
+
+1. **Send the wrong key**
+
+   Try to access the service with the wrong key:
+   
+   ```sh
+   curl -i http://localhost:8000/mock/request \
+     -H 'apikey:bad-key'
+   ```
+  
+   You will receive an an unauthorized response:
+ 
+   ```text
+   HTTP/1.1 401 Unauthorized
+   ...
+   {
+     "message":"Invalid authentication credentials"
+   }
+   ```
+
+1. **Send a valid request**
+
+   Send a request with the valid key in the `apikey` header:
+
+   ```sh
+   curl -i http://localhost:8000/mock/request \
+     -H 'apikey:top-secret-key'
+   ```
+
+   You will receive a `200 OK` response.
+   
+### Service based key authentication
+
+The Key Authentication plugin can be enabled for specific services. The request is the same as above, but posted
+to the service URL:
+
+   ```sh
+   curl -X POST http://localhost:8001/services/example_service/plugins \
+     --data name=key-auth
+   ```
+### Route based key authentication
+
+The Key Authentication plugin can be enabled for specific services. The request is the same as above, but posted
+to the service URL:
+
+   ```sh
+   curl -X POST http://localhost:8001/routes/mocking/plugins \
+     --data name=key-auth
+   ```
+
+## (Optional) Disable the plugin
+
+If you are following this getting started guide section by section, you will need to use this API key 
+in any requests going forward. If you don’t want to keep specifying the key, disable the plugin before moving on.
+
+
+1. **Find the Key Authentication plugin ID**
+
+   ```sh
+   curl -X GET http://localhost:8001/plugins/
+   ```
+   
+   You will receive a JSON response that contains the `id` field, similar to the following snippet:
+   
+   ```text
+   ...
+   "id": "2512e48d9-7by0-674c-84b7-00606792f96b"
+   ...
+   ```
+
+1. **Disable the plugin**
+
+   Use the plugin id obtained above, to `PATCH` the `enabled` field on the plugin:
+
+   ```sh
+   curl -X PATCH http://localhost:8001/plugins/<plugin-id> \
```

Comment:
```suggestion
   curl -X PATCH http://localhost:8001/plugins/example_plugin_id \
```

Reply:
In my opinion, `example_plugin_id` isn't as helpful.  Elsewhere in the guide we use names like this but we want the user to run the commands with them. However, I also agree that assuming a user will know that `<plugin-id>` means to fill in the value may not be clear either.  Above we provide a UUID example from the `GET` request, we could make this example use that UUID instead and ensure we highlight that it's expected the user fill in their value.

Another option is to craft one or two commands that automates this for the user.  We could add `jq` to the getting started guide as a prereq, and then show how to load the ID into a variable or run it as one command.  Or we might be able to process the JSON w/ standard posix tools and help the user obtain the ID value that way.

Reply:
I like the idea of using the UUID in the above example.

Reply:
updated to use UUID